### ISSUE TITLE
Escalation notifications are randomly sent/not sent

### ISSUE THREAD

**Issue by btassite** [centreon/centreon#5706]
CES 3.4.1 iso updated to Centreon 2.8.13, Engine 1.8.0, Broker 3.0.9.

Two issues:
1) timing(?) issue with escalation notifications randomly sent/not sent
2) escalation notifications sent even for unselected states in Service Escalation Options, but not always

How to reproduce:
Create a host, add a service to it with Check and Notification periods 24x7, Max Check Attempts 1, Normal Check Interval 1, Retry Check Interval 1, Notification Interval and First/Recovery notification delay fields empty (same for template and underlying check), Notification Type w,u,c,r.

Create an escalation with First/Last notification set to 1, Notification interval 0, Escalation Period 24x7, Services Escalation Options w,c,r (note that "u" is not checked).

Expected behaviour:
- service status changing from OK to any other state: notification
- escalation notification for all of the above status changes except to unknown.

Observed behaviour:
- service status changing from OK to any other state: notification (OK)
- escalation notification sometimes sent for change to unknown state too, but not always
- escalation notification sometimes NOT sent for changes to states other than unknown

To underscore the biggest issue, I kept changing the output of the test check from OK to CRITICAL and back without touching any of the Centreon config, half the time the escalation notification is sent, and half the time it's not.

**Comment by lpinsivy** [comment]
The escalations are used to repeat the notification every X minutes as long as the problem persists.

Setting the "Notification interval" to 0 is not a valid configuration.

**Comment by btassite** [comment]
0 in Notification Interval for services is valid, which is what all of our setup uses: a single notification is generated when a service enters a Hard state. We are trying to prevent a flood of repeating notifications, so setting it to anything else is unacceptable in our environment.

Using an escalation together with the above however results in undefined behaviour as stated above, i.e. extra notifications randomly being generated or not.

Setting Notification Interval to non zero for both service and escalation (or non zero/zero) works, but since there is no facility to limit the number of service notifications that means a never ending stream of notifications.

The escalation questionmark help text for Notification Interval says: "If you specify a value of 0 for the interval, Monitoring Engine will send the first notification when this escalation definition is valid, but will then prevent any more problem notifications from being sent out."

What I'm trying to achieve: a single escalation notification going out when the service changes (Hard) state from OK to Critical for example (preferably delayed, see #5704). This randomly does and does not happen for escalations in the above scenario. The fact that it's random means there is a bug/race condition somewhere, or if this is a symptom of undefined behaviour due to the input being illegal then the UI should not accept it and the help text should be amended with the conditions under which it is illegal.

"Is volatile" is set to no in the underlying generic-active-service template.

**Comment by lpinsivy** [comment]
Yes define 0 for "First Notification' in escalation definition bypass regular service notification definition.

So if you want to use first service notification definition before escalation, you have to start with '1' in your escalation definition.

If you want to send only one notification in your escalation, you have to define '2' for 'Last Notification'. If you define '0' for 'Last Notification' with X for 'Notification Interval ', your contacts linked to escalation will receive a notification every X minutes until the services is in a non-OK status.

**Comment by btassite** [comment]
It would be nice if the escalation settings did not cause undefined behaviour if you **must** use Notification Interval 0 for service notifications. Will you at least adapt the UI to filter out the invalid configuration as appropriate?
